### PR TITLE
cmd/gocharm: build without 'src' directory, -no-source flag

### LINKED ISSUE #33: New feature: build without 'src' directory
It would be nice to build a charm without 'src' directory. If I build my project with gocharm the resulting directory looks like this:
```bash
11M		bin
12K		config.yaml
172K	hooks
12K		metadata.yaml
12K		README.md
86M		src
```

However the source directory takes 86 MB and isn't really necessary for the charm. Without it the charm is in total around 11 MB. If I remove it and upgrade the charm it works perfectly without it. So do we really need this directory?

If you like I can donate some time for a PR to add a build flag. Copying the 'assets' (https://github.com/juju/gocharm/blob/ee7a68ecffe8444456e86ed1b48a8c8942ed3c89/cmd/gocharm/main.go#L213) folder if there is one.

The reason why my source directory is so large is because it also has some files that are not compiled into the binary. But none the less it seems like a nice feature to have.

### PR DESCRIPTION
**Pull request by mever**
Resolves #33

### CODE REVIEW

File: cmd/gocharm/main.go [hunk 1]
```diff
@@ -184,6 +188,9 @@ func main1(pkgPath string) error {
 		return errgo.Mask(err)
 	}
 	for name := range allowed {
+		if name == "src" && *noSource && !*source {
```
Comment (rogpeppe) [inline cmd/gocharm/main.go]: Given that if *noSource is true, *source must be false, I think this is equivalent to:

```
if *noSource && name == "src" {
    continue
}
```

(I also swapped the two exprs as I think it reads more nicely that way)

File: cmd/gocharm/main.go [hunk 2]
```diff
@@ -117,6 +118,9 @@ func main1(pkgPath string) error {
 	if err != nil {
 		return errgo.Notef(err, "cannot get current directory")
 	}
+	if *source && *noSource {
```
Comment (rogpeppe) [inline cmd/gocharm/main.go]: All the other usage checking is directly in main, so I'd do this check there too, I think.

**Review by rogpeppe — CHANGES_REQUESTED** [review]
LGTM modulo a couple of minor suggestions, thanks!

(and so sorry it's taken 9 months for me to notice this PR)

**Comment by mever** [comment]
Hi @rogpeppe, do not worry about the 9 months. It took me almost a year. :sweat_smile: (I am very busy and this is a side project for me). Anyway, I found some time to throw at this project again.

**Pull request closed without merging**